### DISCUSSION TITLE
Cannot mutate reactive prop [comment] in component: [post-comments]

### DISCUSSION THREAD

**Discussion by PterodactylStore** [livewire/livewire#6033] [7. Bugs (archived)]
### `Cannot mutate reactive prop [comment] in component: [post-comments]`

![image](https://github.com/livewire/livewire/assets/133764150/8b1bb374-64f9-4fcf-ad46-196111e901f8)

`livewire/show-post.blade.php`
```blade
...
@forelse($post->comments as $comment)
    <livewire:post-comments wire:key="{{ $comment->id }}" :$comment />
@empty
    <p class="text-gray-600 dark:text-gray-400">There are no comments</p>
@endforelse
...
``` 
`App\Livewire\PostComments.php`
```php
...
class PostComments extends Component
{
    #[Reactive]
    public $comment;

    public function render()
    {
        return view('livewire.post-comments');
    }
}
...
``` 
`livewire/post-comments.blade.php`
```blade
<div>
    <div class="bg-white dark:bg-gray-700 p-4 my-4 shadow-lg rounded-lg">
        <div class="text-gray-800 dark:text-white">
            <strong>{{ $comment->user->name }}</strong>
            <p>{{ $comment->content }}</p>
        </div>
        <div class="text-gray-600 dark:text-gray-400 text-sm mt-2">
            {{ $comment->created_at->diffForHumans() }}
        </div>
    </div>
</div>

```

**Comment by joshhanley** [comment]
@PterodactylStore thanks for reporting! This behaviour is correct, reactive properties in the child component cannot be modified. If you need to update it, either fire an event to the parent or access the parent directly using `$parent` to update it.

Hope this helps!

**Reply by ghabriel25** [reply]
Hi @joshhanley I think its because the model includes **loaded relations**, so read-only `$comment->user` in render changes the hash and triggers `CannotMutateReactivePropException`. Also `valueMatch()` guard `json_encode === false` while `BaseReactive` does not.

**Comment by robsontenorio** [comment]
@PterodactylStore did you solved this?

@joshhanley @lrljoe  Could, you please, point where `$comment` is being mutated on `post-comments.blade.php`?

**Reply by Emmanuel-Tyty** [reply]
passing :$comment as an array solves the error  for me

**Reply by joshhanley** [reply]
@robsontenorio hmm not sure exactly. Can you make a https://wirebox.app demo so we can have a look?

**Comment by kriit24** [comment]
Make own Reactive Attribute class and override dehydrate method

```
<?php

namespace App\Livewire;

#[\Attribute]
class Reactive extends \Livewire\Attributes\Reactive
{
    //
    public function dehydrate($context)
    {
        //die('sees');

        $context->pushMemo('props', $this->getName());
    }
}

```

**Comment by ghabriel25** [comment]
@PterodactylStore @robsontenorio This has been fixed in https://github.com/livewire/livewire/pull/10536
